### PR TITLE
reverse the output sequence in backwards RNN

### PR DESCRIPTION
**Pull request by RDTm**
When using the parameter "go_backwards" in RNN, we reverse the order of the input sequence going into the layer. We do not, however, reverse the output. In the case of "return_sequences" it means that the first element in output sequence corresponds actually to the last element of the original input sequence. 

This poses following the problems (with "return_sequences"=True):

-  when working with backward RNN we need to reverse the sequence of labels/target values as well. It is not intuitive and not documented anywhere, so its an easy thing to not notice.

-  when wanting to construct a bi-LSTM (or any bidirectional RNN) using a graph model and "go_backward" to create backward branch, we notice that the backward and forward branches will return sequences in opposite order. I did not find any layer allowing to reverse the outputs and make things right. This means it is not possible to construct a sequence-returning bi-LSTM.

Such sequence-returning bi-LSTMs can be useful in, for example: i) gene expression data or ii) when predicting the sentiment of each individual word based on its left and right context.

### CODE REVIEW

**Comment by farizrahman4u** [comment]
I think this should be taken care of at the layer level. (See #1456 / Wrappers/Bidirectional). On seeing the argument `go_backwards` most developers would intuitively think of theano's scan api, and assume that `rnn()` behaves exactly like `scan()`, so its better to leave it that way.

**Comment by farizrahman4u** [comment]
For implementing a bidirectional LSTM with `return_sequences=True`, you can use a `Lambda` layer to reverse your sequence.

```python

def reverse_func(x):
    import keras.backend as K
    rev = K.permute_dimensions(x, (1, 0, 2))[::-1]
    return K.permute_dimensions(rev, (1, 0, 2))

reverse = Lambda(reverse_func) # Add this layer after your go backwards LSTM

```

**Comment by fchollet** [comment]
I agree with @farizrahman4u, I think this is only relevant for building bidirectional RNNs, and therefore should be handled explicitly by the user (e.g. with a `Lambda`). It the general case such a behavior may be counter-intuitive.

**Comment by farizrahman4u** [comment]
@fchollet Reopen?

**Comment by farizrahman4u** [comment]
@fchollet Reversing the output seems to be necessary, otherwise the sequence and the mask will not be aligned as we have discussed before in #1703.

@RDTm Spaces around `=`.

**Comment by danieldk** [comment]
Note though that the code fails on Tensorflow (non-1 steps are not supported):

https://github.com/tensorflow/tensorflow/blob/adc6a4b1463b5bbfceede319b4dd7985947ffefb/tensorflow/python/ops/array_ops.py#L138

So I guess the backend needs an extra function that wraps tf.reverse?

https://www.tensorflow.org/versions/master/api_docs/python/array_ops.html#reverse

**Comment by djstrong** [comment]
Reverse with TensorFlow will be helpful to implement BDRNN.
